tests/snuba/search/test_backend.py

```diff
@@ -1217,6 +1229,297 @@ def test_first_release_environments(self):
+
+        # expect no groups even though there are releases; since no group
+        # is attached to any releases
+
+        results = self.make_query(
+            search_filter_query='first_release:%s' % release_1.version,
+        )
+        assert set(results) == set([])
+
+        results = self.make_query(
+            search_filter_query='first_release:%s' % release_2.version,
+        )
+        assert set(results) == set([])
+
+        # set group_a's first_release to be release_1
+
+        group_a.first_release = release_1
+        group_a.save()
+
+        results = self.make_query(
+            search_filter_query='first_release:%s' % release_1.version,
+        )
+        assert set(results) == set([group_a])
+
+        # set group_b's first_release to be release_1
+
+        group_b.first_release = release_1
+        group_b.save()
+
+        results = self.make_query(
+            search_filter_query='first_release:%s' % release_1.version,
+        )
+        assert set(results) == set([group_a, group_b])
+
+        # set group_c's first_release to be release_2
+
+        group_c.first_release = release_2
+        group_c.save()
+
+        results = self.make_query(
+            search_filter_query='first_release:%s' % release_2.version,
+        )
+        assert set(results) == set([group_c])
+
+        # group_a occured in staging environment, and the latest release at the time
+        # of this occurrence was release_1
+
+        group_a_staging_env = GroupEnvironment.get_or_create(
+            group_id=group_a.id,
+            environment_id=staging_env.id,
+        )[0]
+
+        group_a_staging_env.first_release = release_1
+        group_a_staging_env.save()
+
+        # group_a occured in production environment, and the latest release at the time
+        # of this occurrence was release_2
+
+        group_a_prod_env = GroupEnvironment.get_or_create(
+            group_id=group_a.id,
+            environment_id=prod_env.id,
+        )[0]
+
+        group_a_prod_env.first_release = release_2
+        group_a_prod_env.save()
+
+        # query by release release_1
+
+        results = self.make_query(
+            search_filter_query='first_release:%s' % release_1.version,
+        )
+        assert set(results) == set([group_a, group_b])
+
+        results = self.make_query(
+            environments=[staging_env, prod_env],
+            search_filter_query='first_release:%s' % release_1.version,
+        )
+        assert set(results) == set([group_a])
+
+        results = self.make_query(
+            environments=[staging_env],
+            search_filter_query='first_release:%s' % release_1.version,
+        )
+        assert set(results) == set([group_a])
+
+        results = self.make_query(
+            environments=[prod_env],
+            search_filter_query='first_release:%s' % release_1.version,
+        )
+        assert set(results) == set([])
+
+        # query by release release_2
```

Comment:
@markstory I've grouped the tests for release `release_2` here as outline in the example comment.